### DISCUSSION TITLE
Config File for options not-recommended in devcontainer config

### DISCUSSION THREAD

**Discussion by jckimble** [devcontainers/community#37] [Ideas]
I'm surprised there's not a config file for the cli. There's alot of options that shouldn't/can't be set in a devcontainer.json
The most specific use case for me personally would be to set the dotfiles options without having to set cli options everytime I use something else besides vscode. So the biggest use case would be for users that want to use devcontainers without vscode. I believe this would be a good addition with https://github.com/devcontainers/cli/issues/59 along with giving a alternative option to https://github.com/devcontainers/cli/issues/29

There's a list of options that could be added into this list but the ones I see on my personal usecase are workspace-folder, and dotfiles-repository. But just about any option that has to be set in vscode settings.yml should have an config file option.

**Comment by bamurtaugh** [comment]
Thanks for sharing your scenario here. cc @chrmarti for awareness

**Reply by jckimble** [reply]
No problem. Wasn't sure if I posted in the right place. I have no problem doing this myself and submitting a pull request. From what I saw the cli uses yargs and it's nothing special to call the config function of the library. The part that would take the longest would be filtering out options that don't want to be set by global config

**Comment by jckimble** [comment]
Adding a note to extend use case: would be good to add features also into config file for users that have custom dotfiles that use features.
